basic/airdrop-with-fixed-tokens-by-recipient/src/lib.rs

```diff
@@ -0,0 +1,68 @@
+use scrypto::prelude::*;
+
+
+blueprint! {
+
+    struct AirdropWithFixedTokensByRecipient {
+        admin_badge: ResourceDef,
+        recipients: Vec<Address>,
+        tokens_by_recipient : Decimal
+    }
+
+    impl AirdropWithFixedTokensByRecipient {
+        
+        pub fn new(tokens_by_recipient : Decimal) -> (Component, Bucket) {
+            let admin_badge = ResourceBuilder::new_fungible(DIVISIBILITY_NONE)
+                                .initial_supply_fungible(1);
+            let component = Self {
+                admin_badge : admin_badge.resource_def(),
+                recipients :  Vec::new(),
+                tokens_by_recipient : tokens_by_recipient
+            }
+            .instantiate();
+
+            return (component, admin_badge)
+        }
+
+        pub fn get_minimal_required_tokens(&self) -> Decimal {
+            let recipients_count = self.recipients.len();
+            return Decimal::from(recipients_count) * self.tokens_by_recipient; 
+        }
+
+        #[auth(admin_badge)]
+        pub fn add_recipient(&mut self, recipient: Address) {
+                self.recipients.push(recipient);    
```

Comment:
You should add a check to make sure the recipient is not already inside the `self.recipients` vec. You can use `assert!`

Reply:
Ok, I will make it